pilot/pkg/model/jwks_resolver.go

```diff
@@ -197,9 +210,14 @@ func (r *jwksResolver) GetPublicKey(jwksURI string) (string, error) {
 	}
 
 	// Fetch key if it's not cached, or cached item is expired.
+	// The cache expiration happens if the jwt public key refresh have been constantly failing for
+	// the past 1 week (JwtPubKeyExpireDuration).
+	// For security reasons, we will stop using the cached public key if it still fails this time.
```

Comment:
I'm not sure what security benefit to invalidate the key. It's the public key, so if you use the old (hence invalid key), it is equivalent with using empty key. I'll just return the key from cache for simplicity.

Reply:
This is for the worst case: we failed to refresh the key due to network error for the whole past week.

We will always return the cached key within the 1 week window, regardless of network error. But if it consistently fails for 1 week, we return the empty key. It's different from using the old key, if we return the old key, then the old token will still be accepted.

The key questions, do we want to keep using the old public key *forever* even if we failed to refresh it due to network error? @liminw

Reply:
I feel the decision keeping the key for 1 week is arbitrary, at the same time, doesn't have clear benefit in term of security (happy to see if I'm wrong). Dropping this expiry time and have a *refresher* be responsible for keeping entry up-to-date could make the logic simpler.

Reply:
Anyway, it's ok if you want to keep this expiry logic, but need to make the documentation a bit clearer.

Reply:
This is just to make sure we won't reuse the public key forever which is not a good security practice asI heard from @liminw and @wenchenglu, we can discuss about the value, but I think what you propose is effectively to  make this duration to an infinite greater number?

We already have made the change to the refresher job to update the cached jwt public key periodically, this PR makes it independent from the expire time.